### PR TITLE
Dark mode

### PR DESCRIPTION
**Pull request by biafra23**
Handles #612 and #617

Changes proposed in this pull request:
- Using the purpose instead of a color name in layouts and styles. Example: `error` instead of `tomato` or primary instead of safe_green. Color names are still there and error points to tomato in day mode and coral in night mode
- the color names are generated by Zeplin
- The colors purpose is derived from https://zpl.io/a8oJLjw

description | day | night
----------- | --- | ----
1 Start Screen | Screenshot_20201202_190420 | Screenshot_20201202_191129
2 T&C | Screenshot_20201202_190638 | Screenshot_20201202_191137
3 Load first safe | Screenshot_20201203_094706 | Screenshot_20201203_094715
4 App Settings | Screenshot_20201202_201706 | Screenshot_20201202_191208
5 Import owner Key | Screenshot_20201202_201728  | Screenshot_20201202_191217
6 Select owner key | Screenshot_20201202_201849 | Screenshot_20201202_191312
7 Key imported | Screenshot_20201202_201855 | Screenshot_20201202_191317
8 Load Safe | Screenshot_20201202_201919 | Screenshot_20201202_191330
9 ENS found | Screenshot_20201202_201937 | Screenshot_20201202_191349
10 Add safe bottom sheet | Screenshot_20201202_201927 | Screenshot_20201202_211231
11 Confirm safe selection | Screenshot_20201202_201946 | Screenshot_20201202_191354
12 enter safe name | Screenshot_20201202_201952 | Screenshot_20201202_191359
13 safe name entered | Screenshot_20201202_202000 | Screenshot_20201202_191409
14 Safe Settings | Screenshot_20201202_202015 | Screenshot_20201202_191422
15 Safe Settings 2 | Screenshot_20201202_202027 |  Screenshot_20201202_191426
16 advanced safe settings | Screenshot_20201202_202035| Screenshot_20201202_191430
17 tx list  | Screenshot_20201202_202047 | Screenshot_20201202_191446
18 tx details | Screenshot_20201202_202055 | Screenshot_20201202_191455
19 tx details 2 | Screenshot_20201202_202112 | Screenshot_20201202_191507
20 action | Screenshot_20201202_202153 | Screenshot_20201202_193113
21 advanced tx details  | Screenshot_20201202_202209 | Screenshot_20201202_193133
22 Share safe | Screenshot_20201202_202218 | Screenshot_20201202_193149
23 add safe | Screenshot_20201202_202226 | Screenshot_20201202_193159
24 Creation details | Screenshot_20201202_202251 | Screenshot_20201202_193216
25 Safe Settings  | Screenshot_20201202_202326 | Screenshot_20201202_193231
26 Advabnce Safe Settings  | Screenshot_20201202_202332 | Screenshot_20201202_193237
27 Removing safe conformation | Screenshot_20201202_202346 | Screenshot_20201202_193245
28 Removing owner Key Dialog | Screenshot_20201202_201715 | Screenshot_20201202_211248
29 Confirmation needed | Screenshot_20201202_211847|  Screenshot_20201202_211819
30 Confirmation Dialog | Screenshot_20201202_211851 | Screenshot_20201202_211829

@gnosis/mobile-devs

### CODE REVIEW

File: app/src/main/res/values/colors.xml [hunk 1]
```diff
@@ -1,42 +1,55 @@
...
-    <color name="whitesmoke_two">#f0efee</color>
-
-    <color name="white">#ffffff</color>
+    <color name="toolbar_action">@color/text_emphasis_high</color>
+
+    <color name="background">@color/white</color>
+    <color name="surface_01">@color/white_two</color>
+    <color name="surface_08">@color/white_two</color>
+    <color name="surface_04">@color/white_two</color>
+<!--    <color name="surface_04_dimmed">@color/white_two_dimmed</color>-->
```
Comment (elgatovital) [inline app/src/main/res/values/colors.xml:16]: please remove commented out items

**Review by elgatovital — APPROVED** [review]

File: app/src/main/java/io/gnosis/safe/ui/settings/app/NightModeAppSettingsFragment.kt [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import androidx.appcompat.app.AppCompatDelegate.*
+import androidx.navigation.fragment.findNavController
+import io.gnosis.safe.ScreenId
+import io.gnosis.safe.databinding.FragmentSettingsAppNightModeBinding
+import io.gnosis.safe.di.components.ViewComponent
+import io.gnosis.safe.ui.base.fragment.BaseViewBindingFragment
+import javax.inject.Inject
+
+class NightModeAppSettingsFragment : BaseViewBindingFragment<FragmentSettingsAppNightModeBinding>() {
+    override fun screenId() = ScreenId.SETTINGS_APP_ADVANCED
```
Comment (elgatovital) [inline app/src/main/java/io/gnosis/safe/ui/settings/app/NightModeAppSettingsFragment.kt:16]: please ask if we need to use separate screen id here and update https://github.com/gnosis/safe/wiki/Overview-screen-tracking if necessary

File: app/src/main/java/io/gnosis/safe/ui/settings/app/NightModeAppSettingsFragment.kt [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            }
+            dark.setOnClickListener {
+                darkChecked()
+                settingsHandler.nightMode = MODE_NIGHT_YES
+                settingsHandler.applyNightMode(MODE_NIGHT_YES)
+            }
+        }
+    }
+
+    private fun FragmentSettingsAppNightModeBinding.darkChecked() {
```
Comment (elgatovital) [inline app/src/main/java/io/gnosis/safe/ui/settings/app/NightModeAppSettingsFragment.kt:64]: extension function for binding looks strange. I would just use with(binding) inside a normal private function

File: app/src/main/java/io/gnosis/safe/ui/settings/app/SettingsHandler.kt [hunk 4]
```diff
@@ -2,15 +2,31 @@ package io.gnosis.safe.ui.settings.app
...
+    @NightMode
+    var nightMode: Int
+        get() = preferencesManager.prefs.getInt(KEY_NIGHT_MODE, MODE_NIGHT_FOLLOW_SYSTEM)
+        set(value) {
+            preferencesManager.prefs.edit {
+                putInt(KEY_NIGHT_MODE, value)
+            }
+        }
+
+    fun applyNightMode(@NightMode nightMode: Int) {
```
Comment (elgatovital) [inline app/src/main/java/io/gnosis/safe/ui/settings/app/SettingsHandler.kt:26]: can we combine setting nightMode and applying it into one step (move apply to set(value) of nightMode). They are used always together anyway

Comment (biafra23) [reply app/src/main/java/io/gnosis/safe/ui/settings/app/SettingsHandler.kt:26]: In BaseActivity.onCreate() it is not applied via the setter:
`settingsHandler.applyNightMode(settingsHandler.nightMode)`

File: app/src/main/java/io/gnosis/safe/ui/settings/app/NightModeAppSettingsFragment.kt (hunk 3 above)
Comment (biafra23) [reply app/src/main/java/io/gnosis/safe/ui/settings/app/NightModeAppSettingsFragment.kt:64]: I found it fancy :-) But I don't mind. I used Extract method and this was what AS gave me.

**Pull request merged**